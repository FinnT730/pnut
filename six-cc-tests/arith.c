/* args: abc def hij */
int main(int argc, char_ptr argv) {
  int x;
  int y;
  int z;

  x = 42;
  y = 128;

  z = ++x;
  printf("x: %d, y: %d, z: %d\n", x, y, z);
  z = --x;
  printf("x: %d, y: %d, z: %d\n", x, y, z);
  z = x++;
  printf("x: %d, y: %d, z: %d\n", x, y, z);
  z = x--;
  printf("x: %d, y: %d, z: %d\n", x, y, z);
  z = x += y;
  printf("x: %d, y: %d, z: %d\n", x, y, z);
  z = x -= y;
  printf("x: %d, y: %d, z: %d\n", x, y, z);
  z = x *= y;
  printf("x: %d, y: %d, z: %d\n", x, y, z);
  z = x /= y;
  printf("x: %d, y: %d, z: %d\n", x, y, z);
  z = x %= y;
  printf("x: %d, y: %d, z: %d\n", x, y, z);
}
