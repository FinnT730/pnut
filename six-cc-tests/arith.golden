x: 43, y: 128, z: 43
x: 42, y: 128, z: 42
x: 43, y: 128, z: 42
x: 42, y: 128, z: 43
x: 170, y: 128, z: 170
x: 42, y: 128, z: 42
x: 5376, y: 128, z: 5376
x: 42, y: 128, z: 42
x: 42, y: 128, z: 42
